db/BaseService/index.js

```diff
@@ -0,0 +1,57 @@
+const assert = require("assert");
+const { knex } = require("../helpers");
+
+class BaseService {
+    hasRecord(object) {
+        assert.ok(typeof object === "object", "object<object> is required");
+
+        return knex(this.name)
+            .where(object)
+            .select("id")
+            .then((rows) => {
+                knex.destroy();
+
+                return Boolean(rows.length);
+            });
+    }
+
+    fetchRecordById(recordId) {
+        assert.ok(typeof recordId === "number", "recordId<number> is required");
+
+        return knex(this.name)
+            .where("id", recordId)
+            .select()
+            .first()
+            .then(row => new this.Model(row))
+            .then((model) => {
+                knex.destroy();
+
+                return model;
+            });
+    }
+
+    fetchRecordsBy(object) {
+        assert.ok(typeof object === "object", "object<object> is required");
+
+        return knex(this.name)
+            .where(object)
+            .then(rows => rows.map(row => new this.Model(row)))
```

Comment:
Instead of using the constructor of the Model, I think it is better to have separate methods for turning the model object to db record and vice versa. These methods can under the hood do exactly use the constructor and return all the properties on the object, but it allows you to add methods to the model without them being written to the database and also possible to change the name in the code without changing the name in the database.

Reply:
Not exactly sure what you mean, I'll speak to you about this tomorrow.

Reply:
not sure if we talked about this. But with this code the db model has bled into the the constructor and the field in the database and in the model object always has to be the same.